### PR TITLE
[Enhancement] Send an email when avatar of a user is removed

### PR DESCRIPTION
**Pull request by navaneethkp36**
* Add notify_user_of_avatar_removal action in AvatarsMailer

* Add new form for avatar removal reason and confirmation in app/views/users/edit.html.erb

* Add view for notify_user_of_avatar_removal mail

* Add notify_user_of_avatar_removal deliver_later call in user_controller update action

* Add specs for notify_user_of_avatar_removal action in AvatarsMailer

* Add en.yml text for remove_avatar_confirm_button and remove_avatar_reason

### CODE REVIEW

File: WcaOnRails/app/mailers/avatars_mailer.rb [hunk 1]
```diff
@@ -12,4 +12,17 @@ def notify_user_of_avatar_rejection(user, reason)
...
+  def notify_user_of_avatar_removal(remover_user, user, reason)
+    @remover_user = remover_user
+    @user = user
+    @reason = reason
+
+    mail(
+      from: Team.wrt.email,
+      to: user.email,
+      reply_to: 'results@worldcubeassociation.org',
+      subject: "Your avatar has been removed by #{@remover_user.name}",
```
Comment (Jambrose777) [inline WcaOnRails/app/mailers/avatars_mailer.rb:25]: Need some clarity:
Who is this email going to? WRT and The user who removed the avatar? Or wrt and the user whose avatar was removed?

I think wrt would prefer it just be internal (the person who removed it and wrt), in which case “your” in the subject should probably be changed to the wca id of the user  whose avatar was removed.

Comment (SAuroux) [reply WcaOnRails/app/mailers/avatars_mailer.rb:25]: Please remove remover_user.name from the subject and also do not include this information anywhere else!

WCA teams act as an entity, exposing actions of individual team members  to public entities should never happen if not strictly necessary.

Comment (navaneethkp36) [reply WcaOnRails/app/mailers/avatars_mailer.rb:25]: hi @Jambrose777 @SAuroux 

The email is going to the user whose avatar was removed and yes it includes the name of the user who removed the avatar.

If you check the issue description - https://github.com/thewca/worldcubeassociation.org/issues/7300#issue-1410442641, it clearly states `When an avatar is removed, an email notification to the user with who removed the avatar will be really helpful`.

Is this not the required behavior? I am confused now🤔.

Comment (SAuroux) [reply WcaOnRails/app/mailers/avatars_mailer.rb:25]: > Is this not the required behavior? I am confused now🤔.

Sorry about the confusion, I see how the description was misleading. But what I stated about individual team members definitely needs to be taken into account. Ideally, the notification should distinguish the following two cases:

Case 1: removal by a WRT member
- subject includes "by WRT"
- do **not** include 'results@worldcubeassociation.org' in CC
Case 2: removal by someone else 
- subject includes "by #{@remover_user.name}"
- include 'results@worldcubeassociation.org' in CC

Could you change the implementation accordingly? Thanks in advance!

Comment (navaneethkp36) [reply WcaOnRails/app/mailers/avatars_mailer.rb:25]: @SAuroux sure. Thanks for the clarification. I will add the changes👍

Comment (navaneethkp36) [reply WcaOnRails/app/mailers/avatars_mailer.rb]: Hi, @SAuroux @Jambrose777  I have pushed the changes. Can you please check?

File: WcaOnRails/app/assets/stylesheets/avatars.scss [hunk 2]
```diff
@@ -37,3 +37,16 @@
     background: rgba(0, 0, 0, 0.5);
   }
 }
+
+#upload-avatar-form{
```
Comment (danieljames-dj) [inline WcaOnRails/app/assets/stylesheets/avatars.scss:41]: Nitpick: `#upload-avatar-form {` (having a space before braces) might be better to match other rules of the file.

File: WcaOnRails/app/views/avatars_mailer/notify_user_of_avatar_removal.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<p>Hello <%= @user.name %></p>
```
Comment (danieljames-dj) [inline WcaOnRails/app/views/avatars_mailer/notify_user_of_avatar_removal.html.erb:1]: Shouldn't there be a comma at the end of this line?

File: WcaOnRails/app/views/avatars_mailer/notify_user_of_avatar_removal.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+<p>Your avatar has been removed by <%= @remover %>.</p>
+
+<% if @reason.present? %>
+  <p>
+    Here's the reason:
+    <blockquote><%= @reason %></blockquote>
+  </p>
+<% end %>
+
+<p>Please provide a more suitable picture.</p>
```
Comment (danieljames-dj) [inline WcaOnRails/app/views/avatars_mailer/notify_user_of_avatar_removal.html.erb:12]: A signature at the end would be nice, with something like:
<p>
  Regards,
  WCA Results Team
</p>

**Review by danieljames-dj — CHANGES_REQUESTED** [review]
Hi @navaneethkp36, thanks a lot for working on this and sorry for the late review. I've sent some minor comments. Everything else looks good to me.

File: WcaOnRails/app/views/avatars_mailer/notify_user_of_avatar_removal.html.erb (hunk 3 above)
Comment (navaneethkp36) [reply WcaOnRails/app/views/avatars_mailer/notify_user_of_avatar_removal.html.erb]: @danieljames-dj resolved in 0b2f9e39bedd207f183104d5b7a6151b79560a43🚀

File: WcaOnRails/app/views/avatars_mailer/notify_user_of_avatar_removal.html.erb (hunk 4 above)
Comment (navaneethkp36) [reply WcaOnRails/app/views/avatars_mailer/notify_user_of_avatar_removal.html.erb:12]: @danieljames-dj resolved in 0b2f9e39bedd207f183104d5b7a6151b79560a43🚀

**Review by danieljames-dj — COMMENTED** [review]
Thanks @navaneethkp36. LGTM.
Though, I couldn't review `WcaOnRails/app/views/users/edit.html.erb` much because I didn't understand the changes. Can someone else (@gregorbg) do the review for this file alone (or fully if possible) and approve it?

File: WcaOnRails/app/views/users/edit.html.erb [hunk 5]
```diff
@@ -222,10 +222,23 @@
...
+              </div>
+
+              <div class="remove-avatar-confirm">
+                <%= form_with model: @user, method: :patch do |f| %>
+                  <%= f.text_area :removal_reason, class: 'form-control',
+                    placeholder: t('.remove_avatar_reason') %>
+
+                  <%= f.hidden_field :remove_avatar, value: true %>
+
+                  <%= f.submit t('.remove_avatar_confirm_button'), class: 'btn btn-danger remove-avatar-confirm__button',
```
[the remark this answers is not in the document]
Comment (navaneethkp36) [reply WcaOnRails/app/views/users/edit.html.erb:238]: Thanks for the suggestions @gregorbg . If I am correct these are the 2 implementations that you have mentioned. I personally prefer the second one. What do you think? Please let me know if I am wrong or missing something here.

```css
.remove-avatar-confirm * {
  margin-top: 5px;
}

.remove-avatar-confirm {
  display: none;
}
```
or

```css
.remove-avatar-confirm {
  display: none;
  margin-top: 5px;

  .btn {
    margin-top: 5px;
  }
}
```

[the remark this answers is not in the document]
Comment (gregorbg) [reply WcaOnRails/app/views/users/edit.html.erb:238]: You can choose either one as long as you don't artificially create new HTML classes to target as selectors. But the second version still has the flaw of (seemingly?) duplicating the `margin-top` value...

File: WcaOnRails/app/mailers/avatars_mailer.rb [hunk 6]
```diff
@@ -12,4 +12,18 @@ def notify_user_of_avatar_rejection(user, reason)
...
+
+  def notify_user_of_avatar_removal(remover_user, user, reason)
+    @remover = remover_user.results_team? ? 'the WCA Results Team' : remover_user.name
+    @user = user
+    @reason = reason
+
+    mail(
+      from: Team.wrt.email,
+      to: user.email,
+      cc: ('results@worldcubeassociation.org' unless remover_user.results_team?),
```
[the remark this answers is not in the document]
Comment (navaneethkp36) [reply WcaOnRails/app/mailers/avatars_mailer.rb]: Resolved in https://github.com/thewca/worldcubeassociation.org/pull/7378/commits/eb1c380c09db8d906f6e24f9243560161b72ab61

File: WcaOnRails/app/mailers/avatars_mailer.rb [hunk 7]
```diff
@@ -12,4 +12,18 @@ def notify_user_of_avatar_rejection(user, reason)
...
+      reply_to: 'results@worldcubeassociation.org',
```
[the remark this answers is not in the document]
Comment (navaneethkp36) [reply WcaOnRails/app/mailers/avatars_mailer.rb]: Resolved in https://github.com/thewca/worldcubeassociation.org/pull/7378/commits/4a4057022caeb544c45581e9b82e3cc3f54b919c

File: WcaOnRails/app/assets/stylesheets/avatars.scss [hunk 8]
```diff
@@ -37,3 +37,16 @@
...
+  display: inline;
+}
+
+.remove-avatar-confirm {
+  display: none;
+  margin-top: 5px;
+
+  &__button{
+    margin-top: 5px;
+  }
```
[the remark this answers is not in the document]
Comment (navaneethkp36) [reply WcaOnRails/app/assets/stylesheets/avatars.scss]: Resolved in https://github.com/thewca/worldcubeassociation.org/pull/7378/commits/4a4057022caeb544c45581e9b82e3cc3f54b919c

File: WcaOnRails/app/views/users/edit.html.erb (hunk 5 above)
[the remark this answers is not in the document]
Comment (navaneethkp36) [reply WcaOnRails/app/views/users/edit.html.erb]: Resolved in https://github.com/thewca/worldcubeassociation.org/pull/7378/commits/4a4057022caeb544c45581e9b82e3cc3f54b919c

**Review by gregorbg — APPROVED** [review]

**Review by danieljames-dj — APPROVED** [review]

**Comment by danieljames-dj** [comment]
Thanks, @navaneethkp36 for your contribution. Due to some unknown issues, the test didn't trigger so I pushed an empty commit to your branch and that triggered the test and it succeeded. Merging it now.

**Pull request merged**